obspy/io/sac/sactrace.py

```diff
@@ -1092,7 +1092,7 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                 val = _ut._clean_str(val, strip_whitespace=False)
                 if val.startswith(native_str('-12345')):
                     val = HD.SNULL
-                hs[i] = val
+                hs[i] = val.encode('ASCII', 'replace')
```

Comment:
The other thing I didn't know is that NumPy does an implicit `encode` here when assigning to `hs[i]` (even on Python 3, probably for backwards compatibility reasons, but it's kind of unfortunate.)